### DISCUSSION TITLE
Copilot Support?

### DISCUSSION THREAD

**Discussion by bioshazard** [abetlen/llama-cpp-python#908] [Q&A] [answered]
This documentation suggests LCP can be used with **GitHub Copilot** if you have an OpenAI compatible endpoint.

https://github.com/abetlen/llama-cpp-python/blob/main/docs/server.md#code-completion

I have a working server endpoint, but I get prompted to sign in and later it says I don't have a subscription. 

Does it actually work with Copilot? Do you need to have a subscription for it to work? Does anyone have this working with LCP?

**Comment by abetlen** [comment, marked as answer]
And you updated your `.vscode/settings.json` as suggested?

This might be an annoying limitation of the plugin I haven't run into because I also have a copilot subscription.

**Comment by bioshazard** [comment]
Ah! ok so I do indeed need the sub to to past the "you need a sub" wall. I'll close this as answered ty

**Reply by abetlen** [reply]
Ideally no, I wasn't aware that the extension is that locked down. If so I'd be happy to add support for some more OSS friendly extensions / endpoints like the llm-vscode in #843 or similar.

**Reply by bioshazard** [reply]
TIL llm-vscode. Was going to try to work through cursor.so to attempt using LCP again. But #843 would answer my root interest ty for the link.

**Reply by bioshazard** [reply]
Got LCP working with continue.dev btw, lookin forward to lm-vscode support too

**Comment by hpxiong** [comment]
I suppose this is still not working directly as a drop-in replacement for GitHub Copilot, right? I don't see the endpoint gets called at all after following all steps. And I do not have a Copilot subscription. Continue works but not quite good as Copilot. If this is the case, the documentation probably needs an update.

Another problem is that llama.cpp has problem loading replit-code  Q4 & Q8 gguf models. Only f16 works for me.

**Reply by bioshazard** [reply]
Been meaning to try LCP with cursor.sh since they added OpenAI endpoint override support
